### ISSUE TITLE
What happened ?

### ISSUE THREAD

**Issue by Golem642** [h4h13/paisa-app#2]
The app is no longer accessible on the @RetroMusicPlayer account, what happened to it @h4h13 ?
I really need it it's the best app for this, and a lot of people (myself included) use it everyday

**Comment by h4h13** [comment]
Hi 👋🏻 we decided to make Paisa private repository from now onwards, since to avoid copies

**Comment by Golem642** [comment]
> Hi 👋🏻 we decided to make Paisa private repository from now onwards, since to avoid copies

What copies ? Everyone knows you have the real repo, you own the Play Store app.

And if by copies you mean forks, that's the whole point of GitHub but you can find the original source of a fork anytime easily.

And even then there has been no blatant copy of your app, this is very unjustified and makes me believe this account was actually hacked (why "we" ? Are multiple people using this account ?), which would explain the growth in size + the change of key + making the repo private so you can sneak in a way to exfiltrate data or something.

This isn't any type of app we're talking about, this is a "bank account transactions" type of app which requires the whole thing to be built on trust because it contains sensitive data, but your behavior lately and the weird actions you've taken are worrying some people that this might not be good intentions.

The whole point about open source is you show what you make to everyone, you give them trust about what you are making, and they can easily report problems to you and help you on some of them. Your reason for privating the repo is to avoid "copies", but there is something called a license which you can define what can people do and not do with your app without having to resort to privating the entire repo.

I really don't get it, this decision was so abrupt and left without any notice for anyone.

Please, give us more details on why you chose to private the repo other than just to avoid "copies" which doesn't make sense at all

Best regards

**Comment by chathura-de-silva** [comment]
I gonna quit using the app. Never knew the source code was made private.

**Comment by Golem642** [comment]
@h4h13 we can't even help translate now, plus you have a license as mentioned that you could change anytime if you wanted at the bottom of your readme. I really don't get it, please explain to us why this sudden decision

**Comment by h4h13** [comment]
@Golem642  I don't want Paisa app become like Retro Music Player so I don't want to be opensource anymore because, I barely get support from adding features or bug fixing from being open 

we're gonna add people to repo in private who can support the project in terms of translations or any other means and gives proper credits to supporters

**Comment by h4h13** [comment]
@chathura-de-silva  by all means stop using project if you trust

**Comment by Golem642** [comment]
> @h4h13 said : barely get support from adding features or bug fixing from being open

There were people willing to help you if you just looked at the issues comment, for example on the storage permissions and access for backup.

Besides, what you get from being open is the full trust from the people that uses your app, especially since this is basically a bank account transaction logging app which contains private informations.

In fact you probably gain less from being private than open, people can't help you directly now

**Comment by skj-skj** [comment]
I think the situation is @h4h13 is not making money on this project and some scums on google play publishing the app as paid and got good downloads, they didn't respected the licence.
and its unfair expectations on developers and maintainer to keep working on this project without monetory benifits.

better approach could be to change licence terms or change to source available licence or use [fsl licence](https://fsl.software/), so the project can have a better case when a take down notice issued.

for monetization ads can be an option but will ruin the user experience.
one way of monetization I can think of is have in app purchases on certain features like export to xlsx with charts, tags, auto completed or early access to new features etc. and keep the free app two, three features behind.
other way to monetize by themes, like dark, black or any other fancy theme can be in app purchases.

my speculation is paisa-app going to be integrated with fintech app or will become one, nothing wrong in that if user agreed to new terms and fits their use case.

I have recommended this app to various of my friends who asked me what app to use to track expenses and not sure if I am able to recommend it now.

for me I won't be comfortable, since repo is private and decision to make a closed project is been made I don't know whether my transaction data will be used for monetary gain or not.

and I won't be updating the app in future until there is better clarity on the situation, right now I am using v6.2.3

I hope this is just a bump in the journey of paisa app and things will resolved.

**Comment by FosRexx** [comment]
If anyone is trying to switch from Paisa like I did then I have made a [python script](https://github.com/FosRexx/paisa-to-cashew) that converts the Paisa json backup into [Cashew](https://github.com/jameskokoska/Cashew) import file.

I have been using Cashew for a week now and I think it's great, and it's open source as well.

**Comment by chathura-de-silva** [comment]
> If anyone is trying to switch from Paisa like I did then I have made a [python script](https://github.com/FosRexx/paisa-to-cashew) that converts the Paisa json backup into [Cashew](https://github.com/jameskokoska/Cashew) import file.
> 
> I have been using Cashew for a week now and I think it's great, and it's open source as well.

Thank you for the recommendation and the script.

**Comment by Weldarvin** [comment]
> If anyone is trying to switch from Paisa like I did then I have made a [python script](https://github.com/FosRexx/paisa-to-cashew) that converts the Paisa json backup into [Cashew](https://github.com/jameskokoska/Cashew) import file.
> 
> I have been using Cashew for a week now and I think it's great, and it's open source as well.

I converted mine through excel power query and switch to Cashew 2 weeks ago.

I kinda understand the situation with paisa but still think this could have been handled differently than close-sourcing.

But hey, it's their app and they can do what they want with it.

**Comment by alba4k** [comment]
agree, It's sad that this is no longer open source. Also, no copies have ever been created so I find this quite unjustifiable

**Comment by skj-skj** [comment]
> agree, It's sad that this is no longer open source. Also, no copies have ever been created so I find this quite unjustifiable

It seems that this app is a copy of the [Paisa App](https://play.google.com/store/apps/details?id=premium.quality.moneymanager). Since the Paisa App was under a GPLv3 License, redistribution is allowed, even with monetization. However, the terms of GPLv3 require that they also open-source their code under the same license, which hasn’t been done here.

This issue isn’t unique to the Paisa App, many popular open-source projects face similar challenges. For example, GIMP has been sold on the Microsoft Store, but not by its official maintainers.

**Comment by alba4k** [comment]
You're right, that app is indeed a clone

**Comment by chathura-de-silva** [comment]
> > agree, It's sad that this is no longer open source. Also, no copies have ever been created so I find this quite unjustifiable
> 
> It seems that this app is a copy of the [Paisa App](https://play.google.com/store/apps/details?id=premium.quality.moneymanager). Since the Paisa App was under a GPLv3 License, redistribution is allowed, even with monetization. However, the terms of GPLv3 require that they also open-source their code under the same license, which hasn’t been done here.
> 
> This issue isn’t unique to the Paisa App, many popular open-source projects face similar challenges. For example, GIMP has been sold on the Microsoft Store, but not by its official maintainers.

I did report it. They should have made the codebase available.

**Comment by Golem642** [comment]
> > agree, It's sad that this is no longer open source. Also, no copies have ever been created so I find this quite unjustifiable
> 
> It seems that this app is a copy of the [Paisa App](https://play.google.com/store/apps/details?id=premium.quality.moneymanager). Since the Paisa App was under a GPLv3 License, redistribution is allowed, even with monetization. However, the terms of GPLv3 require that they also open-source their code under the same license, which hasn’t been done here.
> 
> This issue isn’t unique to the Paisa App, many popular open-source projects face similar challenges. For example, GIMP has been sold on the Microsoft Store, but not by its official maintainers.
> 

Omg good eye, i reported it too

@h4h13 I see what you mean now, but please make it open source again and change the license to disallow paid copies of your app. If there's any fake copy that is a problem, publish an announcement and we'll make sure to help you take it down.

We're asking you if you can reconsider your choice, I promise you won't regret it, the community is here to help you

**Comment by Rafee-M** [comment]
> If anyone is trying to switch from Paisa like I did then I have made a [python script](https://github.com/FosRexx/paisa-to-cashew) that converts the Paisa json backup into [Cashew](https://github.com/jameskokoska/Cashew) import file.
> 
> I have been using Cashew for a week now and I think it's great, and it's open source as well.

Paisa's CSV export didn't label expenses correctly and this fixes the issue, thanks!

**Comment by WreckingBANG** [comment]
https://github.com/Kadu-S-Oliveira/Paisa

You can still get the source code from 1 month ago here to compile it yourself.

**Comment by Golem642** [comment]
> https://github.com/Kadu-S-Oliveira/Paisa
> 
> You can still get the source code from 1 month ago here to build it yourself.

This one is even more recent : https://github.com/aleckrh/Paisa

**Comment by chathura-de-silva** [comment]
Well now the intentions and the direction are clearer. He has introduced ads.

**Comment by WreckingBANG** [comment]
@chathura-de-silva Probably making quick money.

**Comment by alba4k** [comment]
Ads have been there for a month and can be disabled with a toggle in the settings

also, they're placed in a way that should make them the least visible possible (e.g. in the settings or when you scroll down a bit)

I don't like the app not being open source anymore either and think that @h4h13 should seriously consider bringing the source code back, but still, y'all should read [this](https://t.me/app_paisa/3602/13856)

**Comment by Golem642** [comment]
I really don't understand, if the ads can be turned off there's no reason to close the apps source code. Besides, the open source repo could probably get some sponsors which mean Hemants gets to have monthly money for it
Also the community could help developing those featured that requires time investment, or at least fix bugs while Hemants is developing those features, it would be a huge time gain
I just hope it's not a question of being overwhelmed with things

**Comment by rosenclosed** [comment]
> so I don't want to be opensource anymore

@h4h13 You know that is in violation of the license terms you chose yourself for this project?

**Comment by skj-skj** [comment]
@rosenclosed previous release will be still comes under gpl.
if there are other contributors other than @h4h13 in the project, they should also agree on this decision. if they disagree on the licence change he have to revert those commits. as the contribution was made under gpl and can't be mix with closed source unless its specified in additional licences.

https://softwareengineering.stackexchange.com/questions/98774/can-gpl-licensed-code-be-close-sourced-later-by-the-author

**Comment by MrAn0nym** [comment]
> if there are other contributors other than @h4h13 in the project, they should also agree on this decision. if they disagree on the licence change he have to revert those commits. as the contribution was made under gpl and can't be mix with closed source unless its specified in additional licences.

There have been (including @h4h13) a total of 39 contributors. While most of them admittedly only provided translations those are nonetheless licensed under gpl… Complaining about people quote-on-quote stealing oss and the ignoring the gpl license and contributor's copyright is extremely ironic

**Comment by Golem642** [comment]
It's been two months since it happened, please answer us @h4h13. We need you, and we need this app

**Comment by h4h13** [comment]
This project will be closed source because of multiple reasons 

1. Source code was recompiled and published as their own apps against the GPLv3
2. Source code was selling on different platforms 
3. I want to implement new features such as cloud back-up and others and improve the project much better but don't want my time wasted by copy cats 😞 
4. Never got any PR which has major features implemented most of the PRs are language translations 


*If developers/ langauge translators want to contribute they can still do it but it will be a private repo

**Comment by Golem642** [comment]
For the record, if anything is against the GPLv3 or any other license you chose, you can file a complaint and they will get taken down for sure.

We'll make sure any copycats stay out of the question, we're your community, if someone is wrongly doing stuff we can help you fix/report it.

As for the PRs, well I can't say that wasn't true. But completely ditching the public PR system rather than having the possibility of something major being shared is probably not the best option.
As an idea, you could start sharing what you plan to do and people would maybe try and help you on these implementations, you just need to give them a guideline on what they can help you with.

I understand your points, but really if anything goes wrong like people stealing your app without respecting the license, then you can just tell us and we'll help you stop it.
Heck, you could probably change the license to a less permissive one if you want but still have the repo public, so that way people can't publish it against your will for sure and you get to keep the full control while being open.

Sometimes the most extreme solutions aren't the best, but if you truly are final about being closed, then I will respect your choice 🫂

**Comment by h4h13** [comment]
`For the people who are saying translations done by the public can't be used in a private repo, sure then I'll remove those translations. Whoever wants to contribute, let them contact as per their consent, will add.`

@Golem642 I don't want to make it open source again. Keeping it private makes me interested in developing the app. I don't want the Paisa app to turn out like RetroMusicPlayer where no help from the community or whatsoever was done.

**Copies**
https://play.google.com/store/apps/details?id=premium.quality.moneymanager

**Code selling**
https://codecanyon.net/item/financial-expense-manager-flutter-app/46210743
https://codecanyon.net/item/takshal-modern-money-manager-app-flutter-hive-android-ios/41903851 


I wanted to help the community with what I have learned in Flutter, Clean Architecture, and other development norms from this project. Even If it helped single person I'm happy for it

**Comment by alba4k** [comment]
Maybe add me to the repo, I was still interested to check out the code sometime (I added italian translations a few years back)

**Comment by Golem642** [comment]
I could handle French translations if nobody else is doing it now

**Comment by h4h13** [comment]
@alba4k  & @Golem642  we have translations files added to this repo you guys are welcome to update and create a PR 
follow the untranslated.json file to see any leftover translations 

https://github.com/h4h13/paisa-app/tree/main/localization
https://github.com/h4h13/paisa-app/blob/main/untranslated.json

**Comment by MrAn0nym** [comment]
@h4h13 I strongly disagree with your views on open source software, I however don't have the time and energy left for a detailed discussion.
That being said, please for God's sake fix, that one can't create transfers. If you want to take our ability to fix this ourselves, do so, but at least keep Paisa in a usable state.

**Comment by BurnoutDV** [comment]
You guys are scum. All I can see is that you wanted to earn money of the work of other people and now wine that your app doesn't sell.

> Never got any PR which has major features implemented most of the PRs are language translations

So you EXPECT others to work for you. Oh no, people only help with translations, something that costs actual companies big money. So what exactly was your plan, to write something "open source", hope others do big chunks of the work and then sell it? Why even going open source in the first place?

I am really angry, I tested various apps, Paisa fit the bill, I wrote a conversion tool for AndroMoney and actually came here to report some shortcomings that might be easily fixeable. But no, the first thing I see is that you are scumbags that violate everything open source stands for. Screw you! Who cares if others make money of it. Its about creating something for everyone, ideally together but with no expectations. If it is too much work then don't do it!
By the way, obscuring the backup json as base64 which actually inflates the files is pathetic.
